lib/containers/urls.pm

```diff
@@ -342,4 +299,28 @@ sub get_3rd_party_images {
     return (\@images);
 }
 
+# Return url of specified image as a string.
+# Specify released => 0 for unreleased image.
+# Die when nothing to return.
+sub get_image_url {
+    my ($self, %args) = @_;
+    $args{version} //= get_required_var('VERSION');
+    $args{arch} //= get_required_var('ARCH');
+    $args{distri} //= get_required_var('DISTRI');
+    $args{released} //= 1;    # 0 => unreleased
+    $args{die} //= 1;    # 0 => do not die
+
+    $args{version} =~ s/^Staging:(?<letter>.)$/Tumbleweed/ if is_tumbleweed || is_microos("Tumbleweed");
+
+    my $url = trim(get_var('CONTAINER_IMAGE_TO_TEST'));
+    return $url unless $url;
+
+    if (supports_image_arch($args{distri}, $args{version}, $args{arch})) {
+        return $images_list{$args{distri}}{$args{version}}{released}->($args{arch}) if ($args{released});
+        return $images_list{$args{distri}}{$args{version}}{totest}->($args{arch});
+    }
```

Comment:
```suggestion
   my $repo = ($args{released}) ? 'released' : totest;
   return $images_list{$args{distri}}{$args{version}}{$repo}->($args{arch});  if supports_image_arch($args{distri}, $args{version}, $args{arch});
```

Reply:
Sorry, I will not do that. It just decreases readability as the lines would be too long.

Reply:
Sorry i reopen it for further discussion. its a small thing but how does it decreases readability? the main issue here is that you have pretty much the same return twice and i cant see any good reason.

Reply:
Readability is the reason.

Reply:
I am not sure I understand. Do you have another suggestion?

Reply:
it depends what do you consider /readability/

Reply:
Agree, that 1 liner is nice but difficult to understand at a first glance...  maybe... 
```
my $type = $args{released} ? 'released' : 'totest';
if supports_image_arch($args{distri}, $args{version}, $args{arch}) {
    return $images_list{$args{distri}}{$args{version}}{$type}->($args{arch})
}
```

Reply:
I like this too, thank you @jlausuch !